src/app/home/addOffer/AddBookOffer.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import styles from "./AddBookOffer.module.css";
+import { useForm } from "react-hook-form";
+import { AddMarketBook, MarketBookStatus } from "@/interfaces/MarketBook";
+import { fetchBookByIsbn } from "@/api/fetchBookByIsbn";
+
+import Input from "@/components/inputs/Input";
+import Select from "@/components/inputs/Select";
+import { useAddBookToMarketMutation } from "@/api/marketApi";
+
+interface FetchStatus {
+  success: boolean;
+  message: string;
+}
+
+export default function AddBookOffer() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<AddMarketBook>();
+  const [formError, setFormError] = useState("");
+  const [fetchStatus, setFetchStatus] = useState<FetchStatus | undefined>(
+    undefined
+  );
+
+  const [addBookToMarket, { isLoading, isSuccess, isError }] =
+    useAddBookToMarketMutation();
+
+  useEffect(() => {
```

Comment:
As in the previous PR, I'd suggest just creating a mapping function instead of using `useEffect` here :)